gax/BUILD.bazel

```diff
@@ -9,6 +9,8 @@ _JAVA_COPTS = [
 
 _COMPILE_DEPS = [
     "@com_google_api_api_common//jar",
+    "@com_google_api_grpc_proto_google_common_protos//jar",
```

Comment:
This is my biggest concern in this PR. We are adding new dependencies to gax. This should be avoided at all cost.

Does this logic work with rest transport? I believe it does not, because unpacking of any should happen much earlier in the call stack in case of rest transport. 

If this logic does not work for rest, please move it to grpc-specific modules and aovid adding extra dependencies to gax.

Reply:
Why should we avoid adding new dependencies to gax at all cost?
Also last time we discussed what should be returned, I thought we said that we should return the error messages object directly.
Most of the logic is in grpc package already, we need it here because `ApiException` need to expose things like `ErrorInfo`. For rest, I would expect that we would convert json to `ErrorInfo`.
However, if we really don't want to add new dependencies, we have two options:
1. Fall back to the String approach I presented before, convert all error message to String first in grpc package, and `ApiException` will only return `List<String>` or `Map<String, String>`.
2. Map all the protobuf models to POJOs in gax, e.g. a new class `ErrorInfo` in gax that mimics the same fields of `ErrorInfo` in protobuf. This will add a significant amount of new classes to gax though since there are ten different message types and some of them have nested types as well.

Reply:
Basically as of now, gax module is independent from protobuf and common protobuf types. It is better to keep it this way. 

On the other hand, looks like both of the newly added dependencies
```
"@com_google_api_grpc_proto_google_common_protos//jar"
"@com_google_protobuf_java//jar"
```
Are already explicitly present in gax-grpc and gax-httpjson. gax is never used without either gax-httpjson or gax-grpc, so looks like it is a low risk - either way in any real setup, both of the new dependencies will be imported for a typical gapic client with or without this change.

At this point it becomes only a conceptual change: gax used to be protobuf-independent module, thought in any practical applicaiton it was always used with protobuf in place (pulled either vie gax-grpc or gax-httpjson). 

I guess, given the practical applications, it is probably ok to introduce this new dependency, even though it is against the original design of gax.